Content.Shared/_RMC14/Deploy/RMCDeploySystem.cs

```diff
@@ -0,0 +1,614 @@
+        if (comp.AreaBlockedCheck && IsAreaBlocked(area, uid, user, ent))
+            return;
+
+        var doAfter = new DoAfterArgs(_entMan, user, comp.DeployTime, new RMCDeployDoAfterEvent(area), uid)
+        {
+            BreakOnMove = true,
+            BreakOnDamage = true,
+            BreakOnHandChange = true,
+            NeedHand = true,
+            AttemptFrequency = comp.AreaBlockedCheck ? AttemptFrequency.EveryTick : AttemptFrequency.Never
+        };
+
+        var started = _doAfter.TryStartDoAfter(doAfter);
+        if (started)
+        {
+            _popup.PopupClient(Loc.GetString("rmc-deploy-popup-start"), ent.Owner, user, PopupType.Small);
+
+            // Sending an event to display the deployment area on the client
+            if (_netManager.IsServer)
+            {
+                var showEvent = new RMCShowDeployAreaEvent(
+                    area,
+                    Color.Blue
+                );
+                RaiseNetworkEvent(showEvent, user);
+            }
+        }
+
+    }
+
+    /// <summary>
+    /// Handles the completion of the deploy do-after event. Deploys setups if not cancelled.
+    /// </summary>
+    private void OnDoAfter(Entity<RMCDeployableComponent> ent, ref RMCDeployDoAfterEvent ev)
+    {
+        if (_netManager.IsClient)
+            return;
+
+        if (ev.Cancelled || ev.Handled)
+        {
+            RaiseNetworkEvent(new RMCHideDeployAreaEvent(), ev.Args.User);
+            return;
+        }
+
+        ev.Handled = true;
+
+        var user = ev.Args.User;
+        // Get the grid and the grid component
+        var gridUid = _xform.GetGrid(user);
+        if (!TryComp<MapGridComponent>(gridUid, out var grid))
+            return;
+
+        // Get the player's world position
+        var userWorldPos = _xform.GetWorldPosition(user);
+        // Tile indices under player
+        var tileIndices = _map.WorldToTile(gridUid.Value, grid, userWorldPos);
+        // The center of this tile (for calculating the zone)
+        var tileCenter = _map.TileCenterToVector(gridUid.Value, grid, tileIndices);
+
+        // Build transform and zone relative to tileCenter
+        var areaTransform = new Robust.Shared.Physics.Transform(tileCenter, 0);
+        var area = ent.Comp.DeployArea.ComputeAABB(areaTransform, 0);
+        var areaCenter = area.Center;
+
+        // Perform deployment
+        DeploySetups(ent, areaCenter, user);
+
+        RaiseNetworkEvent(new RMCHideDeployAreaEvent(), ev.Args.User);
+    }
+
+    /// <summary>
+    /// Deploys setups, considering the possibility of redeploying existing entities.
+    /// </summary>
+    private void DeploySetups(Entity<RMCDeployableComponent> ent, Vector2 areaCenter, EntityUid user)
+    {
+        if (_netManager.IsClient)
+            return;
+
+        // Ensure the original entity has a ContainerManagerComponent
+        EntityManager.EnsureComponent<ContainerManagerComponent>(ent.Owner);
```

Comment:
Use the proxy method
```suggestion
        EnsureComp<ContainerManagerComponent>(ent.Owner);
```